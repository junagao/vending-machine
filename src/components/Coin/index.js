export { default as CoinItem } from './CoinItem';
export { default as CoinList } from './CoinList';
export { default as CoinSlot } from './CoinSlot';
export { default as CollectRefund } from './CollectRefund';
export { default as DisplayDepositedAmount } from './DisplayDepositedAmount';
export { default as RefillCoins } from './RefillCoins';
export { default as Wallet } from './Wallet';
