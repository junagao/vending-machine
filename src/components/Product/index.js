export { default as CollectProduct } from './CollectProduct';
export { default as ProductList } from './ProductList';
export { default as ProductItem } from './ProductItem';
export { default as RefillProductStock } from './RefillProductStock';
