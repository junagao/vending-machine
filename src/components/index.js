export { default as CoinSlot } from 'components/CoinSlot';
export { default as CollectProduct } from 'components/CollectProduct';
export { default as VendingMachine } from 'components/VendingMachine';
export { default as VendingMachineScreen } from 'components/VendingMachineScreen';
export { default as Wallet } from 'components/Wallet';
export { default as ProductList } from 'components/ProductList';
export { default as Header } from 'components/Header';
export { default as CollectRefund } from 'components/CollectRefund';
