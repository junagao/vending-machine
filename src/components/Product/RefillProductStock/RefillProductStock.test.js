import React from 'react';
import { shallow } from 'enzyme';
import { RefillProductStock } from 'components';

describe('RefillProductStock', () => {
  let wrapper;
  const productsMock = [
    {
      id: 'coke',
      name: 'coke',
      img: 'image/coke.png',
      price: 1,
      currentStock: 10,
    },
    {
      id: 'sprite',
      name: 'sprite',
      img: 'image/sprite.png',
      price: 1.25,
      currentStock: 5,
    },
    {
      id: 'fanta',
      name: 'fanta',
      img: 'image/fanta.png',
      price: 1.25,
      currentStock: 11,
    },
  ];

  const onRefillProductStockMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <RefillProductStock
        products={productsMock}
        onRefillProductStock={onRefillProductStockMock}
      />,
    );
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one Card element with prop `title` and `subtitle`', () => {
    const titleMock = 'Refill machine products';
    expect(wrapper.find('Card').length).toEqual(1);
    expect(wrapper.find('Card').prop('title')).toEqual(titleMock);
    expect(wrapper.find('Card').prop('subtitle')).toEqual(`(+5)`);
  });

  it('should render one RefillList element', () => {
    expect(wrapper.find('RefillList').length).toEqual(1);
  });

  it('should render three RefillItem element', () => {
    expect(wrapper.find('RefillItem').length).toEqual(3);
  });

  it('should render three Button element with text prop containing coin name and prop handleClick', () => {
    expect(wrapper.find('Button').length).toEqual(3);
    expect(
      wrapper
        .find('Button')
        .at(0)
        .prop('text'),
    ).toEqual(`coke`);
  });
});
