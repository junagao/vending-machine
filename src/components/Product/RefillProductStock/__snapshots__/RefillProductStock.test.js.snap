// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefillProductStock should render correctly given the required props 1`] = `
<Card
  lastCard={false}
  subtitle="(+5)"
  title="Refill machine products"
>
  <RefillList>
    <RefillItem
      key="coke"
    >
      <Button
        handleClick={[Function]}
        text="coke"
      />
    </RefillItem>
    <RefillItem
      key="sprite"
    >
      <Button
        handleClick={[Function]}
        text="sprite"
      />
    </RefillItem>
    <RefillItem
      key="fanta"
    >
      <Button
        handleClick={[Function]}
        text="fanta"
      />
    </RefillItem>
  </RefillList>
</Card>
`;
